### DISCUSSION TITLE
Any limitation for the log output with the Java executor?

### DISCUSSION THREAD

**Discussion by TomerFi** [sormuras/junit-platform-maven-plugin#89] [Q&A] [answered]
I think I'm seeing the reducted log,
it reaches up to about 500 tests under one container.

**Comment by sormuras** [comment]
Which log do you refer to? Do you want to have/introduce a configurable limit or do you experience some (unwanted) limitations?

**Reply by TomerFi** [reply]
As the saying goes:
"A demo repo with a workflow exemplifying the case is worth 1000 words".
:joy:

[Example repo](https://github.com/TomerFi/junit-plugin-log-cut) with 500 test cases in 1 container.
Note [this workflow's result](https://github.com/TomerFi/junit-plugin-log-cut/runs/4896095825?check_suite_focus=true#step:4:1050) running *mvn test*,
after test *493*, the output is cut,
even the overall tests summary is not displayed.

**Comment by sormuras** [comment]
Now I get it. Thanks for "big picture". 🙃

![image](https://user-images.githubusercontent.com/2319838/150558989-c21a63f3-c5a2-4b14-87aa-602e11525a50.png)
...

![image](https://user-images.githubusercontent.com/2319838/150559104-07372b2c-caa6-4a3e-83df-6265831b5d09.png)

**Comment by sormuras** [comment]
Easiest work-around is to hide the tree and show only the summary: pass `--details=summary` as an extra argument.

I don't recall a limit within JUnit console's code. So, I guess, a max line limit is applied by Maven? For each plugin?

@khmarbaise @rmannibucau any insights regarding my guess?

**Comment by rmannibucau** [comment, marked as answer]
Yes using the plain console has no issue and no maven has no limitation.
I think I would blame junit platform plugin without more knowledge :p (https://github.com/sormuras/junit-platform-maven-plugin/blob/709c6da49665bd5af8817f45c6b817bb0a356acc/src/main/java/de/sormuras/junit/platform/maven/plugin/JavaExecutor.java#L142)

**Comment by TomerFi** [comment]
Yes, I can confirm removing the `limit(500)` works.
Any reason to keep it?

**Reply by rmannibucau** [reply]
guess memory so readAllLines must move to lines API to stream the output instead of loading it all but limit can be dropped

**Reply by TomerFi** [reply]
got it.
thank you very much!
😄

**Comment by TomerFi** [comment]
Fixed in #91
